applets/clock/clock.c

```diff
@@ -2254,6 +2253,7 @@ cities_changed (GSettings    *settings,
 {
         LocationParserData data;
         GSList *cur = NULL;
+        GSList *l;
```

Comment:
You don't need to use a double-linked list since you only use `next` pointer, instead you can use a [GSList](https://developer.gnome.org/glib/stable/glib-Singly-Linked-Lists.html#GSList).

Reply:
Actually, this one already is a singly-linked list :)

Reply:
Oops! ;-)